staking/integration-tests/tests/initialize_pool.rs

```diff
@@ -116,3 +117,51 @@ fn test_update_y() {
         0
     );
 }
+
+#[test]
+fn test_update_reward_program_authority() {
+    let SetupResult {
+        mut svm,
+        payer,
+        reward_program_authority,
+        ..
+    } = setup(SetupProps {
+        init_config:            true,
+        init_target:            true,
+        init_mint:              true,
+        init_pool_data:         true,
+        init_publishers:        true,
+        reward_amount_override: None,
+    });
+
+    let pool_config_pubkey = get_pool_config_address();
+    let pool_config: PoolConfig = fetch_account_data(&mut svm, &pool_config_pubkey);
+
+    assert!(pool_config.reward_program_authority == reward_program_authority.pubkey());
+
+    let new_reward_program_authority = Pubkey::new_unique();
+    update_reward_program_authority(
+        &mut svm,
+        &payer,
+        &reward_program_authority,
+        new_reward_program_authority,
+    )
+    .unwrap();
+
+    let pool_config: PoolConfig = fetch_account_data(&mut svm, &pool_config_pubkey);
+    assert!(pool_config.reward_program_authority == new_reward_program_authority);
+
+    // Trying to update the reward program authority without the correct authority should fail
+    let wrong_authority = Keypair::new();
```

Comment:
You could also test this with `reward_program_authority` instead of generic a new wrong authority since it's no longer the authority

Reply:
done.